src/Parser/TIFFParser.js

```diff
@@ -0,0 +1,67 @@
+import * as UTIF from 'utif';
+import * as THREE from 'three';
+
+/**
+ * The TIFFParser module provides a [parse]{@link module:TIFFParser.parse}
+ * method that takes a TIFF in and gives a `THREE.DataTexture` that can be
+ * displayed in the view.
+ *
+ * It uses the [utif](https://www.npmjs.com/package/utif) library to parse the
+ * TIFF.
+ *
+ * @example
+ * Fetcher.arrayBuffer('image.tiff')
+ *     .then(TIFFParser.parse)
+ *     .then(function _(texture) {
+ *         var source = new FileSource({ parsedData: texture });
+ *         var layer = new ColorLayer('tiff', { source });
+ *         view.addLayer(layer);
+ *     });
+ *
+ * @module TIFFParser
+ */
+export default {
+    /**
+     * Parse a TIFF file and return a `THREE.DataTexture`.
+     *
+     * @param {ArrayBuffer} data - The TIFF file content to parse.
+     *
+     * @return {Promise} A promise resolving with a `THREE.DataTexture`.
+     */
+    parse(data) {
+        const IFD = UTIF.decode(data)[0];
+        UTIF.decodeImage(data, IFD);
+
+        // A 4096 limitation sounds the most probable here
+        // https://webglstats.com/webgl/parameter/MAX_TEXTURE_SIZE
+        if (IFD.width > 4096 || IFD.height > 4069) {
+            throw new Error('The loaded TIFF is too big: please reduce it to a maximum size of 4096 by 4096');
+        }
+
+        // Round to next power of two
+        const width = 2 ** (~~Math.log2(IFD.width - 1) + 1);
```

Comment:
there is utils in THREE to compute `next power of two`.

Reply:
Nice, changed thanks !